File: UI/grooveon_mobile/lib/models/subscription_plan_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'subscription_plan_response.g.dart';

@JsonSerializable()
class SubscriptionPlanResponse {
  final int id;
  final String name;
  final String? planCode;
  final double price;
  final int durationDays;
  final String? description;
  final bool isActive;

  SubscriptionPlanResponse({
    required this.id,
    required this.name,
    this.planCode,
    required this.price,
    required this.durationDays,
    this.description,
    required this.isActive,
  });

  factory SubscriptionPlanResponse.fromJson(Map<String, dynamic> json) =>
      _$SubscriptionPlanResponseFromJson(json);

  Map<String, dynamic> toJson() => _$SubscriptionPlanResponseToJson(this);
}

File: GrooveOn.Models/ResponseObjects/SubscriptionPlanResponse.cs
namespace GrooveOn.Model.ResponseObjects
{
    public class SubscriptionPlanResponse
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string PlanCode { get; set; } = string.Empty;

        public decimal Price { get; set; }

        public int DurationDays { get; set; }

        public string? Description { get; set; }

        public bool IsActive { get; set; }
    }
}
